lib/ansible/modules/packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,400 @@
+                args += " --disable=" + n
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+            return True
+        else:
+            return False
+
+    def disable_all(self):
+        notice("redhat_repository: disable all repos")
+        args = "subscription-manager repos --disable='*'"
+        rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+
+def main():
+    #
+    # Initialise Ansible module
+    #
+    module = AnsibleModule(
+        argument_spec = dict(
+            id      = dict(type="list"),
+            # can't set a default='all' on list, otherwise id will never be used to enable/disable repos
+            list    = dict(choices=['all', 'enabled', 'disabled']),
+            mode    = dict(choices=['idempotent', 'incremental']),
+            state   = dict(choices=['enabled', 'disabled']),
+        ),
+        required_one_of = [['id', 'list']],
+        mutually_exclusive = [['id', 'list']],
+        supports_check_mode = False,
+    )
+
+    #
+    # Initialize RhsmRepositories
+    #
+    rhsmrepos = RhsmRepositories(module)
+
+    p_list = module.params['list']
+    p_id = module.params['id']
+    p_state = module.params['state']
+    p_mode = module.params['mode']
+
+    # setting the default here, because if done at the AnsibleModule declaration, it screws things up
+    if p_mode is None:
+        p_mode='idempotent'
+
+    notice( "redhat_repository: in main | list=" + str(p_list) + " | id=" + str(p_id) + " | state=" + str(p_state) + " | mode=" + str(p_mode) )
+
+    if p_list is not None:
+        notice("redhat_repository: in main, list is not None")
+        if p_list in [ 'all', 'enabled', 'disabled' ]:
+            result=rhsmrepos.list_stuff(p_list)
+            module.exit_json(changed=False, redhat_repositories=result)
+        else:
+            # we shouldn't reach this with Ansible.
+            module.exit_json(changed=False, error="Unknown list primitive")
+
+    elif p_id is not None:
+        notice("redhat_repository: in main, id is not None")
+        has_changed=False
+        changed_repos={}
+
+        if p_mode == 'idempotent':
+            if p_state == 'enabled':
+                ( has_changed, changed_repos ) = rhsmrepos.enable_only(p_id)
+            else:
+                module.exit_json(changed=False, error="Can't use state=disabled in idempotent mode")
+
+        elif p_mode == 'incremental':
+            if p_state == 'enabled':
+                has_changed=rhsmrepos.enablerepo(p_id)
+                changed_repos['enabled'] = p_id
+            elif p_state == 'disabled':
+                has_changed=rhsmrepos.disablerepo(p_id)
+                changed_repos['disabled'] = p_id
+
+        changed_repos.setdefault('enabled', [])
+        changed_repos.setdefault('disabled', [])
+        module.exit_json(changed=has_changed, state=p_state, id=p_id, enabled=changed_repos['enabled'], disabled=changed_repos['disabled'])
```

Comment:
This line is too long even by the Ansible code style guide. It should be under 120 chars per line. Please format it like this:

```
module.exit_json(
    changed=has_changed,
    state=p_state,
    id=p_id,
    enabled=changed_repos['enabled'],
    disabled=changed_repos['disabled'])
```